tests/queries/0_stateless/02354_vector_search_expansion_search.sql

```diff
@@ -0,0 +1,38 @@
+-- Tags: no-fasttest, long, no-asan, no-asan, no-ubsan, no-debug
+-- ^^ Disable test for slow builds: generating data takes time but a sufficiently large data set
+-- is necessary for different hnsw_candidate_list_size_for_search settings to make a difference
+
+-- Tests vector search with setting 'hnsw_candidate_list_size_for_search'
+
+SET allow_experimental_vector_similarity_index = 1;
+SET enable_analyzer = 0;
+
+DROP TABLE IF EXISTS tab;
+
+-- Generate some data set that is large enough 
+CREATE TABLE tab(id Int32, vec Array(Float32), INDEX idx vec TYPE vector_similarity('hnsw', 'L2Distance')) ENGINE = MergeTree ORDER BY id SETTINGS index_granularity = 8192;
+INSERT INTO tab SELECT number, [toFloat32(randCanonical(1)), toFloat32(randCanonical(2))] FROM numbers(500000); -- if the test fails sporadically, increase the table size, HNSW is non-deterministic ...
```

Comment:
Maybe we could use `generateRandom` with a specified seed, so the random values would be deterministic. What do you think?

Reply:
`generateRandom` does not allow to generate a series of integers 1, 2, 3, ... for use as key.

Nevermind, I found an alternative way to generate some deterministic randomess.